examples/access-control-migration/src/services/assign-project-instance-id.voter.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  AuthorizationContext,
+  AuthorizationMetadata,
+  AuthorizationDecision,
+} from '@loopback/authorization';
+import _ from 'lodash';
+
+// Instance level authorizer
+// This is a WORKAROUND to modify the authorization context
+// It is not used for making a decision, so just returns ABSTAIN
+export async function assignProjectInstanceId(
+  authorizationCtx: AuthorizationContext,
+  metadata: AuthorizationMetadata,
+) {
+  // hard-coded for a collection of known endpoints
+  // TBD(FEAT): We should support instance level context for object,
+  // like project1, project2, project3
+  // the current AuthrozationCtx is designed for class level object,
+  // like projects
+  const projectId = authorizationCtx.invocationContext.args[0];
+  const resourceName = getResourceName(metadata.resource ?? authorizationCtx.resource, projectId);
```

Comment:
at the moment `authorizationCtx.resource` is not really used  . wondering if we should remove it.

Reply:
in case metadata doesn't have `resource`, I think we should still keep it.

Reply:
Verified again, it has to be there, because according to the interface [AuthorizationContext](https://github.com/strongloop/loopback-next/blob/master/packages/authorization/src/types.ts#L89) and [AuthorizationMetadata](https://github.com/strongloop/loopback-next/blob/master/packages/authorization/src/types.ts#L56), metadata has `resource` as optional while context has it as required, which means the authorization context guarantees the `resource` is provided.

If I remove it, `getResourceName` has to make the 1st parameter optional:

```ts
function getResourceName(resource: string, id?: number): string {
  // instance level name
  if (id) return `${resource}${id}`;
  // class level name
  return `${resource}*`;
}
```
having multiple optional parameters is annoying, and also doesn't reflect the truth, the resource name should be guaranteed to exist when call the function.